src/components/styledSelect/styledSelect.test.tsx

```diff
@@ -45,12 +47,84 @@ describe('StyledSelect', () => {
         <StyledSelect
           options={options}
           onOptionSelected={() => {}}
+          onSubmitInput={() => {}}
           placeholder="Doesn't matter"
         />
       )
 
       expect(wrapper).toMatchSnapshot()
     })
+
+    describe('submitting the input', () => {
+      describe('when not pressing enter', () => {
+        test("doesn't call the onSubmitInput function", () => {
+          const onSubmitInput = vitest.fn()
+
+          const wrapper = render(
+            <StyledSelect
+              options={options}
+              onOptionSelected={() => {}}
+              onSubmitInput={onSubmitInput}
+              placeholder="Doesn't matter"
+            />
+          )
+
+          const input = wrapper.getByTestId('selectedOption')
+
+          act(() => fireEvent.keyDown(input, { target: { key: ' ' } }))
+
+          expect(onSubmitInput).not.toHaveBeenCalled()
+        })
+      })
+
+      describe('when the input value matches an option name', () => {
+        test.skip('selects the option', () => {
```

Comment:
These tests are failing because of the `node-canvas` issue. Hopefully they will update `node-canvas` soon.